class-wc-gateway-iyzico-checkout-form.php

```diff
@@ -0,0 +1,715 @@
+
+                        $wpdb->update(
+                                $table_name, array(
+                            'customer_id' => $customer_id,
+                            'card_key' => $card_user_key,
+                            'iyzico_api' => $merchant_api_id,
+                                ), array('customer_id' => $customer_id)
+                        );
+                    }
+                }
+
+                $item_transactions = $response->getPaymentItems();
+
+                foreach ($item_transactions as $item_transaction) {
+                    $table_name = $wpdb->prefix . 'iyzico_order_refunds';
+
+                    $wpdb->insert(
+                            $table_name, array(
+                        'order_id' => $response->getBasketId(),
+                        'paid_price' => $item_transaction->getPaidPrice(),
+                        'item_id' => $item_transaction->getItemId(),
+                        'payment_transaction_id' => $item_transaction->getPaymentTransactionId(),
+                        'total_refunded' => 0
+                            )
+                    );
+                }
+
+                $redirect_url = add_query_arg(array('msg' => addslashes($this->msg['message']), 'type' => $this->msg['class']), $checkout_orderurl);
+                wp_redirect($redirect_url);
+                exit;
+            } catch (\Exception $ex) {
+                $respMsg = $ex->getMessage();
+                $respMsg = !empty($respMsg) ? $respMsg : "Invalid Request";
+                wc_add_notice(__($respMsg, 'iyzico-woocommerce-checkout-form'), 'error');
+                $redirect_url = $woocommerce->cart->get_checkout_url();
+                wp_redirect($redirect_url);
+                exit;
+            }
+        }
+
+    }
+
+}
+
+add_filter('woocommerce_payment_gateways', 'woocommerce_add_iyzico_checkout_form_gateway');
+
+function woocommerce_add_iyzico_checkout_form_gateway($methods) {
+    $methods[] = 'WC_Gateway_Iyzicocheckoutform';
+    return $methods;
+}
+
+function iyzico_checkout_form_load_plugin_textdomain() {
+    load_plugin_textdomain('iyzico-woocommerce-checkout-form', FALSE, plugin_basename(dirname(__FILE__)) . '/i18n/languages/');
+}
+
+add_action('plugins_loaded', 'iyzico_checkout_form_load_plugin_textdomain');
+
+class iyzicocheckoutformGateway {
+
+    private $_pluginSettings = array();
+    private $_wcOrder = array();
+
+    function __construct($settings, $order_id) {
+        $this->_pluginSettings = $settings;
+        $this->_wcOrder = new WC_Order($order_id);
+    }
+
+    function generatePaymentToken() {
+        global $wpdb;
+        $valid_currency = array("TRY", "GBP", "USD", "EUR", "IRR");
+
+        if (!in_array($this->_wcOrder->get_order_currency(), $valid_currency)) {
+            $response = __('Please sure the currency value in your payment request is one of the (USD, EUR, GBP, IRR, TL) valid values.', 'iyzico-woocommerce-checkout-form');
```

Comment:
texti "Please make sure that the currency value in your payment request is one of the (USD, EUR, GBP, IRR, TL) valid values." olarak değiştirelim.